runtime/pubsub/internal/utils/utils.go

```diff
@@ -216,7 +218,9 @@ func pointify(rval reflect.Value, ptrDepth int) reflect.Value {
 // configuration in the RetryPolicy
 func GetDelay(maxRetries int, minDelay, maxDelay time.Duration, attempt uint16) (shouldRetry bool, backoff time.Duration) {
 	if maxRetries == types.NoRetries || (int(attempt) > maxRetries && maxRetries != types.InfiniteRetries) {
-		return false, 0
+		// return the max delay here, so if we can't dead letter a message
+		// the could will at least apply the largest allowed backoff
+		return false, maxDelay
```

Comment:
hmmm. is this meant to be used if the deadlettering of a message fails? the original intent with the 0  was no retry.

Reply:
In AWS we can't forcable deadletter something, so we just keep the backoff going until AWS decides to dead letter it